Stevens-EWS/faculty-profiles
# Ek 406 update gatsby to v3

Summary of Changes:

1. Added a `.nvmrc` file and set the version to `14.17.4` (latest stable version).
2. Added `@link` extensions in our schema for the array fields. Resolvers are no longer inferred by default in gatsby v3.
3. Removed createRemoteFileNode.
4. Migrated from using `gatsby-image` to `gatsby-plugin-image` for an updated API for v3.  Removed `gatsby-image` package.  
5. Deleted unused `Image.js` component and updated `Index.js` to not contain all of gatsby's default content that we never took out.

Test:
1. There should be no noticeable changes from what is currently in prod and this build.
2. All array fields are being rendered correctly (this goes with test #1 but this was a major pain point so better to note it in tests)
3. Test caching:
- Deploy with cache clear on Netlify
- Change one of the photos `last_modified` filed on `people-dev2`.
- Deploy without clearing cache
- Check the build log to see if the modified photo is being downloaded again and all others are imported from the cache.

Notes:
1. Remove these two lines `[context.production]
    environment = { PEOPLEAPI_HOST = 'api.stevens.edu/people-dev2/faculty'}` from `netlify.toml` after testing is done.
2. Merge https://bitbucket.org/stevens_devops/people/pull-requests/46/ek-431-tweak-people-api before merging this PR because there are changes here that depend on the changes in the People API.
3. Photos do not work on Cloudflare at the moment and that's because we're using Netlify caching plugin.
